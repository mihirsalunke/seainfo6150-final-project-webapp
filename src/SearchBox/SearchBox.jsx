import React from 'react'
import PropTypes from "prop-types";
import styles from "./SearchBox.module.css";

const SearchBox = (props) => {
    return (
        <div className={styles.searchBox}>
            <input
                className={styles.searchText}
                type='search'
                placeholder='search events...'
                onChange={e => props.onChange(e.target.value)}
            />
            <button className={styles.searchButton} onClick={props.onClick}>Search</button>
        </div>
    );
};

SearchBox.propTypes = {
    onChange: PropTypes.func.isRequired,
};

export default SearchBox;