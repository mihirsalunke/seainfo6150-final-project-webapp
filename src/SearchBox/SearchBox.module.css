@import url("https://fonts.googleapis.com/css2?family=Rowdies:wght@300&display=swap");

.searchBox {
    /* position: absolute;
    top: 50%;
    left: 50%;

    transform: translate(-50%, -50%); */
    background: white;
    height: 40px;
    border: 4px solid black;
    border-radius: 40px;
    padding: 10px; 
    font-family: Rowdies, sans-serif;
}

.searchText {
    border: none;
    background: none;
    outline: none;
    float: left;
    padding: 0;
    color: black;
    font-size: 16px;
    transition: 0.4s;
    line-height: 40px;
    width: 270px;
    font-family: Rowdies, sans-serif;
    font-size: 2em;
}

/* .searchBox:hover > .searchText {
    width: 240px;
    padding: 0 6px;
} */

.searchBox:hover > .searchButton {
    background: orange;
    transition: 0.4s;
}

.searchButton {
    color: black;
    float: right;
    width: 120px;
    height: 40px;
    border-radius: 40px; 
    background: lightgray;
    display: flex;
    justify-content: center;
    align-items: center;
    font-family: Rowdies, sans-serif;
    font-size: 2em;
}