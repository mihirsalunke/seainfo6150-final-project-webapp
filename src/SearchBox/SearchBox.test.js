import React from "react";
import SearchBox from "./SearchBox.jsx";
import { render } from "@testing-library/react";

describe("SearchBox tests", () => {
    it("renders correctly", () => {
      const { container } = render(<SearchBox onChange={() => {}} />);
      expect(container).toMatchSnapshot();
    });
});