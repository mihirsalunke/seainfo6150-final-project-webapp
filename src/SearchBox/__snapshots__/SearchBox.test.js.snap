// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox tests renders correctly 1`] = `
<div>
  <div
    class="searchBox"
  >
    <input
      class="searchText"
      placeholder="search events..."
      type="search"
    />
    <button
      class="searchButton"
    >
      Search
    </button>
  </div>
</div>
`;
